src/time/Waveform.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <Eigen/Core>
+#include "utils/EigenHelperFunctions.hpp"
+
+namespace precice {
+namespace time {
+
+class Waveform {
+public:
+  void addNewWindowData(Eigen::VectorXd data)
```

Comment:
I don't like this name too much. What is the new window? Do we only add data after each window?

Reply:
Currently yes. But this is also related to the discussion [here](https://github.com/precice/precice/pull/1004#discussion_r627166583).

Reply:
We now have two methods `updateThisWindow` and `moveToNextWindow`. Maybe this is better?